geode-log4j/src/main/java/org/apache/geode/logging/log4j/internal/impl/LogWriterAppender.java

```diff
@@ -302,10 +302,15 @@ public synchronized void startSession() {
 
   @Override
   public synchronized void stopSession() {
+    if (logWriter == null) {
+      pause();
+      return;
+    }
     LOGGER.info("Stopping session in {}.", this);
     logWriter.shuttingDown();
     pause();
     logWriter.closingLogFile();
+    logWriter = null;
```

Comment:
I think it would be safer to place `logWriter = null;` within a finally block just in case one of those logWriter calls above throws something like an IOException or a nested IOException.